Marketplace landing

he-is-james opened the pull request:
still need to style but the functionality of the sorting and filtering pop-up is finished

he-is-james committed:
initial creation of produce card

allowed pass through of components for name and price, will work on getting test img

he-is-james committed:
added more prop fields

he-is-james committed:
working navigation of card to details

he-is-james committed:
props passed from card to detail screen

need to figure out prop types because of eslint error, still builds though

he-is-james committed:
started styling produce card

need to add icon and figure out flex to position everything

he-is-james committed:
started airtable integration

can call the table to get produce, will need a way to map the data in produce cards

he-is-james committed:
Merge branch 'master' into marketplace

he-is-james committed:
display airtable data as produce cards

needs to fix initial load with useState delay

he-is-james committed:
fixed initial render and styled produce cards

going to marketplace now loads all the produce from the airtable
styled the produce cards to look like the figma (may need to adjust sizing because they are pixels not percentages)

he-is-james committed:
fixed spelling error

he-is-james committed:
established checks for empty fields in airtable

default values if empty, pass in new image parameters with urls

he-is-james committed:
quick fixes

taken updates to produce card design
adjusted proptypes for detail screen

he-is-james committed:
passed in new information

provided all fields of information needed for the detail screen
began styling

he-is-james committed:
Merge branch 'master' into expanded-produce-card

he-is-james committed:
need to adjust props type

he-is-james committed:
finished number input

removed minimum quantity
updates quantity value with textinput and increment and decrement buttons

he-is-james committed:
styling for detailsScren

he-is-james committed:
detailscreen styling

he-is-james committed:
added favorite passover from thumbnail to details

warning with persistence, but works
need to have it update to airtable

he-is-james committed:
added styling for popup

he-is-james committed:
add to cart adds produce info to cart tbd

includes all fields
will get to duplicate orders and adding user id

he-is-james committed:
add to cart considers update or create

he-is-james committed:
adjusted produce

started favoriting implementation
resized frontend design for smaller phones

he-is-james committed:
created grid view

he-is-james committed:
began subscription structuring

he-is-james committed:
started filter popup

he-is-james committed:
working checkbox

he-is-james committed:
working seasonal filter

he-is-james committed:
implemented filter

he-is-james committed:
started sort

he-is-james committed:
working sort and filter

issue with ordering, not working when currently sorted adds a new filter or when current filters add a new sort

he-is-james committed:
sorting and filtering order fixed

he-is-james committed:
started styling

hwang635 committed:
Merge branch 'master' into marketplace-landing

hwang635 commented on src/Marketplace/MarketplaceScreen.js:
```diff
@@ -45,24 +53,20 @@ const styles = StyleSheet.create({
   const [produceList, setProduceList] = useState([]);
 
-  const getProduce = () => {
+  const [subscriptions, setSubscriptions] = useState(false);
```
Can you remove all of the subscription stuff? Some of it it causing errors (like the subscription button) but generally it just isn't needed anymore.

hwang635 commented on src/Marketplace/MarketplaceScreen.js:
```diff
@@ -92,69 +96,173 @@ export default function MarketplaceScreen({ navigation }) {
         produce.fields.Favorited = false;
```
There might be an issue with the Favorited toggle, I'm getting an unknown field error when I toggle. 
<img width="182" alt="favourite" src="https://user-images.githubusercontent.com/28603722/163326318-cb4aa6c7-23a7-422d-a4ae-23596c5087f0.PNG">
<img width="193" alt="unknown field name favourite" src="https://user-images.githubusercontent.com/28603722/163326349-872d5ebd-7f4f-4058-b8ed-a5d8bda58851.PNG">

hwang635 commented on src/Marketplace/MarketplaceScreen.js:
```diff
@@ -92,69 +96,173 @@ export default function MarketplaceScreen({ navigation }) {
+  const [vegetablesFilter, setVegetablesFilter] = useState(false);
+  const [fruitsFilter, setFruitsFilter] = useState(false);
+
+  const filterProduce = (listToFilter) => {
```
This is a really efficient/concise function

hwang635 commented on src/Marketplace/ProduceGrid.js:
```diff
@@ -0,0 +1,55 @@
+import ProduceCard from './ProduceCard';
+
+const styles = StyleSheet.create({
+  produceCardsContainer: {
```
Good job figuring out how to get everything styled in a grid format

hwang635 commented on src/Marketplace/FilterPopup.js:
```diff
@@ -0,0 +1,106 @@
+import PropTypes from 'prop-types';
+import CheckboxIcon from 'react-native-vector-icons/MaterialCommunityIcons';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+// radiobox-blank
```
Remove these comments unless they mean something

hwang635 reviewed (changes requested):
Hi James, the filtering functionality works super well, I don't see any issues. I'm really impressed that you were able to figure out a relatively tough feature so quickly! There are some bugs I found with the subscription stuff and the favourite icon that are causing crashes, but those should be easy fixes/removals. Thanks for your work on this!

hwang635 commented on src/Marketplace/MarketplaceScreen.js:
```diff
@@ -92,69 +96,173 @@ export default function MarketplaceScreen({ navigation }) {
+            <Button
+              style={styles.button}
+              onPress={() => { setSubscriptions(false); }}
+            >
```
Remove these 2 buttons now that there's no subscription

he-is-james committed:
removal of unnecessary information

he-is-james committed:
adds to airtable

needs to append to list, currently overwrites
need to unfavorite

hwang635 reviewed (changes requested):
Hey James, everything looks good but there have been some updates to airtable that I think is affected your airtable calls and causing errors. Just those fixes and fixing the merge conflict with master and then you're good

he-is-james committed:
fixed add to cart feature

takes in passed user id
adjusted fields and table names

he-is-james committed:
finalized favoriting system

adds to the array if favorited
removes from the array if unfavorited

hwang635 commented on src/Marketplace/ProduceCard.js:
```diff
@@ -74,31 +74,36 @@ function ProduceCard({
+      } else {
+        currentFavorites = currentFavorites.filter((item) => item !== produce);
+      }
+      console.log(currentFavorites);
```
Remove console.log

hwang635 reviewed (changes requested):
This looks pretty good, only thing is that I'm seeing an airtable error when I interact w/ favouriting the apple/carrots sometimes. Is this as an expected error as a result of hardcoding ids or is it an actual bug?

AirtableError {
  "error": "INVALID_VALUE_FOR_COLUMN",
  "message": "Value \"rec3jGoylvKNoEyok,rec2McK3K5xMq45yh,rec2McK3K5xMq45yh\" contains duplicate value. Each record ID must appear only once",      
  "statusCode": 422,
}

he-is-james committed:
working favoriting

updating favorite variable
update airtable

he-is-james committed:
working favorite filtering

he-is-james committed:
final adjustments

used airtable to call for updates to the favorites local array and then also passing to render

he-is-james committed:
fixed filterproduce and showfavorites functions

needed to adjust the order in which favorited items would appear when filters were added since the rerender of a variable could not determine certain situations so another field was added

he-is-james committed:
Merge branch 'master' into marketplace-landing

hwang635 commented on src/Marketplace/MarketplaceScreen.js:
```diff
@@ -65,19 +66,22 @@ export default function MarketplaceScreen({ navigation }) {
+    let favorites = [];
+    await base('Users').find(userId, (err, record) => {
+      if (err) {
+        console.error(err);
```
can you replace all console.errors in all of the screens w/         Alert.alert(err.error, err.message);

hwang635 commented on src/Marketplace/ProduceDetailsScreen.js:
```diff
@@ -137,15 +137,42 @@ const styles = StyleSheet.create({
 
   const [favorited, setFavorited] = useState(favorite);
 
+  const addToFavorites = async (user, produce, favcondition) => {
```
This is a v smart function

hwang635 reviewed (changes requested):
Hi James, this looks really good! Your code is very clean and clever and everything works well, only thing to fix is the error messages.

he-is-james committed:
final adjustments
